src/dagmc.cpp

```diff
@@ -818,12 +864,56 @@ int32_t next_cell(int32_t surf, int32_t curr_cell, int32_t univ)
   return univp->cell_index(new_vol);
 }
 
+extern "C" int openmc_dagmc_universe_get_cell_ids(
+  int32_t univ_id, int32_t* ids, size_t* n)
+{
+  // make sure the universe id is a DAGMC Universe
+  const auto& univ = model::universes[model::universe_map[univ_id]];
+  if (univ->geom_type() != GeometryType::DAG) {
+    fatal_error(
+      "Universe " + std::to_string(univ_id) + " is not a DAGMC Universe!");
+  }
+
+  std::vector<int32_t> dag_cell_ids;
+  for (const auto& cell_index : univ->cells_) {
+    const auto& cell = model::cells[cell_index];
+    if (cell->geom_type() == GeometryType::DAG)
```

Comment:
The cells in this Universe should always be `DAGCell`'s. Maybe this could be a check to ensure that a `CSGCell` isn't found? Otherwise I think we can remove it.